portal/src/hooks/useJournal.ts

```diff
@@ -0,0 +1,279 @@
+import {
+  QueryClient,
+  QueryFunctionContext,
+  useMutation,
+  useQuery,
+  useQueryClient,
+} from "react-query";
+
+import http from "../utils/httpMoonstream";
+import { MOONSTREAM_API_URL } from "../constants";
+import useUser from "../contexts/UserContext";
+import { Entities, Entity, Journal, JournalInput, JournalQueryData, Journals } from "../types";
+
+const DEFAULT_JOURNAL_NAME = "Web3Addresses";
+
+interface EntityCreationInput {
+  address: string;
+  title: string;
+  blockchain: string;
```

Comment:
Add `chainid` (and maybe remove `blockchain`?)

Reply:
Maybe just use `blockchain:<chainID>` instead of `blockchain:polygon`, `blockchain:ethereum`, etc.